src/Umbraco.Web.UI.Client/src/common/services/tinymce.service.js

```diff
@@ -167,6 +167,14 @@ function tinyMceService($rootScope, $q, imageHelper, $locale, $http, $timeout, s
         xhr = new XMLHttpRequest();
         xhr.open('POST', Umbraco.Sys.ServerVariables.umbracoUrls.tinyMceApiBaseUrl + 'UploadImage');
 
+        xhr.onloadstart = function(e) {
+            eventsService.emit("rte.file.uploading");
```

Comment:
these need to be wrapped in a digest else the angular app doens't actually respond since these events are triggered outside of an angular digest. Safest thing to do is wrap this in an `angularHelper.safeApply`. If you don't, you'll notice that the state of the buttons doesn't change when you first start uploading until you move the mouse around, or similar.